dbt/config.py

```diff
@@ -90,27 +90,94 @@ def colorize_output(config):
     return config.get('use_colors', True)
 
 
-def _render(key, value, ctx):
-    """Render an entry in the credentials dictionary, in case it's jinja.
+class ConfigRenderer(object):
+    """A renderer provides configuration rendering for a given set of cli
+    variables and a render type.
+    """
+    def __init__(self, cli_vars):
+        self.context = {'env_var': env_var}
+        self.context['var'] = Var(None, self.context, cli_vars)
 
-    If the parsed entry is a string and has the name 'port', this will attempt
-    to cast it to an int, and on failure will return the parsed string.
+    @staticmethod
+    def _is_hook_path(keypath):
+        if not keypath:
+            return False
 
-    :param key str: The key to convert on.
-    :param value Any: The value to potentially render
-    :param ctx dict: The context dictionary, mapping function names to
-        functions that take a str and return a value
-    :return Any: The rendered entry.
-    """
-    if not isinstance(value, compat.basestring):
+        first = keypath[0]
+        # run hooks
+        if first in {'on-run-start', 'on-run-end'}:
+            return True
+        # model hooks
+        if first in {'seeds', 'models'}:
+            if 'pre-hook' in keypath or 'post-hook' in keypath:
+                return True
+
+        return False
+
+    @staticmethod
+    def _is_port_path(keypath):
+        return len(keypath) == 2 and keypath[-1] == 'port'
+
+    @staticmethod
+    def _convert_port(value, keypath):
+
+        if len(keypath) != 4:
+            return value
+
+        if keypath[-1] == 'port' and keypath[1] == 'outputs':
```

Comment:
i assume you meant to use `_is_port_path` here.

this is ok for now. thought: this should be on the adapter as part of "Adapters as Plugins"